source/Missions/Patch/BoardGames/BoardGamePatches.cs

```diff
@@ -119,13 +99,14 @@ static bool Postfix(bool result)
         }
     }
 
-    [HarmonyPatch(typeof(BoardGameBase), "HandlePlayerInput")]
-    public class HandlePlayerInputPatch
+    [HarmonyPatch(typeof(BoardGameAIBase), "CalculateMovementStageMoveOnSeparateThread")]
+    public class CalculateMovePatch
     {
-        public static event Action<Move> OnHandlePlayerInput;
-        static void Postfix(ref BoardGameBase __instance, ref Move __result)
+        public static bool Prefix()
         {
-            OnHandlePlayerInput?.Invoke(__result);
+            if (!BoardGameLogic.IsPlayingOtherPlayer) return true;
```

Comment:
nit: no need for an if statement, just `return BoardGameLogic.IsPlayingOtherPlayer`?